slsim/Sources/SourceVariability/variability.py

```diff
@@ -64,3 +119,130 @@ def variability_at_time(self, observation_times):
         :return: variability at given time.
         """
         return self._model(observation_times)
+
+    def parse_kwargs_for_lamppost_reprocessed_model(self):
+        """Separates categories of self.kwargs for lamppost reprocessing.
+
+        :return: Dictionaries for:
+            - agn kwargs
+            - driving signal kwargs
+            - reprocessing kwargs
+            - signal kwargs
+        """
+        self.agn_kwargs = {}
+        self.driving_signal_kwargs = {}
+        self.reprocessing_kwargs = {}
+        self.signal_kwargs = {}
+
+        self.redshift = 0
+        self.delta_wavelength = 50
+
+        for kwarg in self.kwargs_model:
+            if kwarg in [
+                "r_out",
+                "corona_height",
+                "r_resolution",
+                "inclination_angle",
+                "black_hole_mass_exponent",
+                "black_hole_spin",
+                "eddington_ratio",
+            ]:
+                self.agn_kwargs[kwarg] = self.kwargs_model[kwarg]
+
+            elif kwarg in ["time_array", "magnitude_array"]:
+                self.signal_kwargs[kwarg] = self.kwargs_model[kwarg]
+
+            elif kwarg in [
+                "length_of_light_curve",
+                "time_resolution",
+                "log_breakpoint_frequency",
+                "low_frequency_slope",
+                "high_frequency_slope",
+                "mean_magnitude",
+                "standard_deviation",
+                "normal_magnitude_variance",
+                "zero_point_mag",
+                "input_frequencies",
+                "input_psd",
+                "light_curve",
+                "seed",
+                "amp",
+                "freq",
+            ]:
+                self.driving_signal_kwargs[kwarg] = self.kwargs_model[kwarg]
+
+            elif kwarg in ["redshift"]:
+                self.redshift = self.kwargs_model[kwarg]
+            elif kwarg in ["delta_wavelength"]:
+                self.delta_wavelength = self.kwargs_model[kwarg]
+            else:
+                self.reprocessing_kwargs[kwarg] = self.kwargs_model[kwarg]
+
+    def reprocess_with_lamppost_model(self):
```

Comment:
This function is also inside the variability class. Is it possible to move this function outside of the variability class. If it is necessary to put this function inside the variability class, I am ok with this.